Add Helix icon (#8421)

kanielrkirby opened the pull request:
<!--
Before opening your pull request, have a quick look at our contribution guidelines:
https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md

Consider adding a preview image of your submission using:
https://wasm.simpleicons.org/preview
-->

**Issue:** closes #8421

**Popularity metric:**

<!--
Regardless of whether or not the linked issue (if there is one) has a metric, please include the metric here for PR reviewers to validate. See our contributing guidelines at https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md#assessing-popularity for more details on how we assess a brand's popularity.
-->

[31.2k GitHub stars](https://github.com/helix-editor/helix/stargazers)

### Checklist

- [x] I updated the JSON data in `_data/simple-icons.json`
- [x] I optimized the icon with SVGO or SVGOMG
- [x] The SVG `viewbox` is `0 0 24 24`

### Description

<!--
Anything relevant, for example:
  - Why did you pick the hex value?
  - Did you manually vectorize the logo?
  - Have you used multiple sources?
  - etc.
-->

- Picked the Hex color from the background of [the website](https://helix-editor.com).
- I took the SVG from [the website](https://helix-editor.com/logo.svg).
- I added gaps, [as discussed](https://github.com/simple-icons/simple-icons/issues/8421#issuecomment-1887537100) in #8421.

uncenter commented:
Please add a preview image as directed by the top comment in the PR template.

uncenter commented:
> Not sure if there it was supposed to be an SVG or something else, but the "Save Preview" and "Copy" buttons on the website fail with `BorrowMut` errors, I assume from Rust/WASM @uncenter. I just took a screenshot of it

I'm unable to reproduce this. Would you mind opening an issue on https://github.com/mondeja/simple-icons-website-rs and make sure to include your browser and browser version details?

kanielrkirby commented on icons/helix.svg:
```diff
@@ -0,0 +1 @@
+<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>Helix</title><path d="M12.604 4.97 4 0v3.55c0 .449.239.863.627 1.087l4.276 2.47zm5.339 3.083-3.7 2.138 4.86 2.807c.11-.19.17-.407.17-.633V9.552c0-.452-.241-.87-.633-1.096zm1.33 12.397c0-.449-.24-.863-.627-1.087l-4.253-2.456-3.7 2.137L19.273 24zm-13.92-4.49 3.7-2.137c-2.703-1.562-4.884-2.82-4.884-2.82a1.26 1.26 0 0 0-.16 ...
```
My bad (Helix adds a trailing newline ironically enough, and Figma doesn't play well by default centering smaller SVGs perfectly), should be fixed now though.

uncenter replied:
> Helix adds a trailing newline ironically enough

Does it not [support EditorConfig](https://github.com/simple-icons/simple-icons/blob/7746c05122cef5cc2b0510b85a41387e27589d16/.editorconfig#L13-L14)?

kanielrkirby replied:
It appears [no, not yet](https://github.com/helix-editor/helix/issues/279).

uncenter replied:
Shoot. I keep trying/checking in on Helix every few months but progress is pretty slow. Neovim it is for now!

kanielrkirby replied:
I've really enjoyed it honestly @uncenter, if you ever get tired of setting up your plugins, Helix comes with basically 95% of what you need from NeoVim without any setup. Even LSPs just need the binaries in your path. But to be honest, the workflow is just different. NeoVim is all about customization, and Helix is about providing sane defaults. Their plugin system's slowly chugging along, and I'm hopeful the more I look at it, it's using Scheme (what TreeSitter uses for queries under the hood).

kanielrkirby replied:
That's my opinion on it anyways :sweat_smile:

uncenter replied:
Thank you for the well thought out explanation! I'll have to give it another try. I definitely prefer the.. how do you say it.. their keybinds/actions setup for modal text editing as opposed to the Vim like motions. Thankfully I've come across a neat Nix project to make customizing Neovim sane ahah: https://github.com/uncenter/flake/blob/main/home/programs/neovim.nix.

kanielrkirby replied:
Nice, glad you appreciate! I was using [NixVim](https://github.com/nix-community/nixvim) for a while actually, if you're still wanting to stick with NeoVim a bit more. It makes setting things up a little easier (so long as you already understand the like 7 layer abstraction cake that is VimScript/Lua/NeoVim/Nix, which luckily I did :sweat_smile: 

But yeah, you should give it a shot, I haven't looked back haha.

kanielrkirby commented:
> > Not sure if there it was supposed to be an SVG or something else, but the "Save Preview" and "Copy" buttons on the website fail with `BorrowMut` errors, I assume from Rust/WASM @uncenter. I just took a screenshot of it
> 
> I'm unable to reproduce this. Would you mind opening an issue on https://github.com/mondeja/simple-icons-website-rs and make sure to include your browser and browser version details?

I just made the issue, though it might just be a garbage issue now that I look into it. It's definitely something to do with canvas / maybe WASM, but it's something screwy I have set up surely, since it happens on Figma too.

uncenter commented:
@kanielrkirby are you still interested in working on this pull request?

kanielrkirby commented:
> @kanielrkirby are you still interested in working on this pull request?

I definitely can, I was under the impression I had addressed your concerns. Is there anything this needs?

uncenter commented:
Ah, my apologies! I misread/remembered the earlier conversation incorrectly 😅. I'll ask another reviewer to take a look here (I'm not super familiar with gaps and all that).

kanielrkirby commented:
> Ah, my apologies! I misread/remembered the earlier conversation incorrectly 😅. I'll ask another reviewer to take a look here (I'm not super familiar with gaps and all that).

Not a problem!

LitoMore reviewed (approved):
Thank you for contributing! This looks pretty good both in large and small sizes!
